### PR TITLE
chore: lock conversations on closed issues and PRs with message

### PR DESCRIPTION
**Pull request by GavinZZ**
### Reason for this change

Lock conversations on closed issues and PRs to make sure important comments are not left on closed items and no one would be able to see it. Force people to create new issues instead.

### Description of changes

Lock conversations.

### Description of how you validated changes

Tested in a forked repo and it works correctly.

### Checklist
- [ ] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/workflows/lock-issue-pr-with-message.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            If you wish to keep having a conversation with other community members under this issue feel free to do so.
+  lock:
+    permissions:
+      pull-requests: write
+      issues: write
+    runs-on: ubuntu-latest
+    needs: auto_comment
+    steps:
+      - name: Lock closed PRs and issues
+        uses: dessant/lock-threads@v3
```
Comment (moelasmar) [inline .github/workflows/lock-issue-pr-with-message.yml:31]: we should not depend on 3rd party GH actions

Comment (GavinZZ) [reply .github/workflows/lock-issue-pr-with-message.yml]: Updated to directly use GH API.

**Comment by moelasmar** [comment]
it looks good to me. The only thing is missed, and I do not think we need to add it as GH workflow is to do lock all old closed issues and PRs, as this workflow will work only for newly closed issues or prs

**Review by moelasmar — APPROVED** [review]

File: .github/workflows/lock-issue-pr-with-message.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      issues: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: aws-actions/closed-issue-message@v1
+        with:
+          repo-token: "${{ secrets.GITHUB_TOKEN }}"
+          message: |
+            Comments on closed issues and PRs are hard for our team to see. 
+            If you need help, please open a new issue that references this one.
+            If you wish to keep having a conversation with other community members under this issue feel free to do so.
```
Comment (moelasmar) [inline .github/workflows/lock-issue-pr-with-message.yml:22]: I think this line should be removed, I think community will not be able to add comments

Comment (moelasmar) [reply .github/workflows/lock-issue-pr-with-message.yml:22]: sorry I missed this line while reviewing the PR

Comment (GavinZZ) [reply .github/workflows/lock-issue-pr-with-message.yml:22]: Oh good point. Let me remove it.

**Pull request merged**